### PR TITLE
Enabling different variants of LCOW, etc.

### PR DESCRIPTION
**Pull request by bknueven**
## Fixes/Addresses:
Part of #402.

## Summary/Motivation:
Sometimes we might want to track different variants of specific energy consumption and/or LCOW; see https://github.com/watertap-org/watertap/pull/393#discussion_r823893782

## Changes proposed in this PR:
- Separating out annual_water_production
- Allowing user-specified names for annual_water_production, LCOW, and specific_energy_consumption so different flow basis can be analyzed
- Adding calls to `add_annual_water_production` where appropriate

## Questions:
- Should we add any more safety checking around these methods? Calling them more than once with the same name is probably a mistake.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/costing/watertap_costing_package.py [hunk 1]
```diff
@@ -252,50 +256,85 @@ def initialize_build(self):
             self.total_operating_cost, self.total_operating_cost_constraint
         )
 
-        if hasattr(self, "LCOW"):
-            calculate_variable_from_constraint(self.LCOW, self.LCOW_constraint)
+        for var, con in self._registered_LCOWs.values():
+            calculate_variable_from_constraint(var, con)
 
-    def add_LCOW(self, flow_rate):
+    def add_LCOW(self, flow_rate, name="LCOW"):
```
Comment (adam-a-a) [inline watertap/costing/watertap_costing_package.py:258]: Can you add a test demonstrating, say, LCOW with respect to feed flow?

**Review by TimBartholomew — APPROVED** [review]
LGTM

**Review by adam-a-a — APPROVED** [review]
LGTM - I guess we don't _need_ tests for the mods, but they wouldn't hurt.

**Comment by bknueven** [comment]
> LGTM - I guess we don't _need_ tests for the mods, but they wouldn't hurt.

Note that the way this is structured, it's more-or-less tested already. I also have a separate checkbox on #402 to work on testing for the costing package, but I was focusing on feature enhancement before moving on to testing and documentation.

**Pull request merged**